web/src/components/Header.js

```diff
@@ -46,19 +101,29 @@ class Header extends Component {
   }
 }
 
-const mapStateToProps = ({ auth: { authenticated } }) => ({
-  authenticated
+const mapStateToProps = state => ({
+  authenticated: state.auth.authenticated,
+  currentUser: state.auth.user,
+  isAdmin: getIsAdmin(state)
 });
 
 const mapDispatchToProps = { pushRoute: push };
 
 Header.propTypes = {
   authenticated: PropTypes.bool.isRequired,
-  pushRoute: PropTypes.func.isRequired
+  currentUser: PropTypes.object,
+  isAdmin: PropTypes.bool.isRequired,
+  pushRoute: PropTypes.func.isRequired,
+  ariaExpanded: PropTypes.bool.isRequired
 };
 
+Header.defaultProps = {
+  currentUser: null,
+  ariaExpanded: false
+};
 
-export default connect(
-  mapStateToProps,
-  mapDispatchToProps
-)(Header);
+export default withRouter(
```

Comment:
I'm using `withRouter` here to give the component access to `location.pathname`, but after looking at `Login.js`, I'm not sure I need to do this?

Reply:
You need `withRouter` if your component is not in a `<Route>`.  The `<Route>` component will provided `location` and `history` props to the thing it renders, so that's why `Login.js` doesn't need it, but `Header.js` probably does.